src/Service/EncryptionService.php

```diff
@@ -56,9 +75,15 @@ public function encryptField($value): EncryptedColumnVO
     {
         if ($value instanceof LazyLoadingInterface) {
             /** @var LazyLoadingInterface|ValueHolderInterface $value */
+            // If the value hasn't been decrypted; it hasn't been changed. Don't bother reencrypting unless it
+            // was encrypted using a different configuration
             if (!$value->isProxyInitialized()) {
-                //if data hasn't been encrypted, we don't need to change it; set it back to what it was at load
-                return $this->originalValues[spl_object_hash($value)];
+                $original = $this->originalValues[spl_object_hash($value)];
+                if (
+                    !$original->needsReencryption($this->encryptor->getIdentifier(), $this->serializer->getIdentifier())
```

Comment:
I was hoping for @paragonie-scott to comment on this; the goal of halite is to make all the parameter choices for the end user; even if this lib was to adopt other libraries I'd look for libs that took the same stance. I'd hope that halite would be able to read data encrypted by previous versions of itself regardless of the parameters in use at the time. Leaving this library responsible only for knowing that halite encrypted it as opposed to say defuse.